File: src/main/java/quebec/virtualite/backend/services/rest/CityDTO.java
package quebec.virtualite.backend.services.rest;

import jakarta.validation.constraints.NotBlank;
import lombok.With;
import quebec.virtualite.backend.services.domain.entities.CityEntity;

public record CityDTO(@NotBlank @With String name, @NotBlank @With String province)
{
    public CityDTO(CityEntity entity)
    {
        this(entity.name(), entity.province());
    }

    public CityEntity toEntity(long id)
    {
        return new CityEntity(id, name, province);
    }
}

File: src/main/kotlin/quebec/virtualite/backend/services/rest/CityDTO.kt
package quebec.virtualite.backend.services.rest

import jakarta.validation.constraints.NotBlank
import quebec.virtualite.backend.services.domain.entities.CityEntity

data class CityDTO(

    @field:NotBlank
    val name: String?,

    @field:NotBlank
    val province: String?,
)
{
    constructor() : this(null, null)

    constructor(entity: CityEntity) : this(entity.name, entity.province)

    fun toEntity(id: Long): CityEntity = CityEntity(id, name!!, province!!)
}
